rust_src/src/frames.rs

```diff
@@ -225,14 +227,19 @@ pub fn framep(object: LispObject) -> LispObject {
 /// use `display-graphic-p' or any of the other `display-*-p'
 /// predicates which report frame's specific UI-related capabilities.
 #[lisp_fn(min = "0")]
-pub fn window_system(frame: Option<LispFrameRef>) -> Option<LispSymbolRef> {
-    let f = frame.map_or(selected_frame(), |f| f.as_lisp_obj());
-    let window_system = framep(f);
+pub fn window_system(frame: Option<LispFrameRef>) -> LispObject {
```

Comment:
I've thought that we changed our return types and now use more explicit types. You didn't complain about this in the last PRs ?